### PR TITLE
Group foreach loops in a single for loop

### PR DESCRIPTION
**Pull request by otapliger**
Just to make it less error prone. The change from 'foreach' to 'for' is because it adds the possibility to add css classes to distinct entries, for example the last entry might have a higher padding-bottom than the others

### CODE REVIEW

File: src/Views/ShortcutsView.vala [hunk 1]
```diff
@@ -26,6 +26,8 @@ public class ShortcutOverlay.ShortcutsView : Gtk.Grid {
     private const string SCHEMA_MEDIA = "org.gnome.settings-daemon.plugins.media-keys";
     private const string SCHEMA_MUTTER = "org.gnome.mutter.keybindings";
 
+    private Gtk.SizeGroup size_group = new Gtk.SizeGroup (Gtk.SizeGroupMode.HORIZONTAL);
```
Comment (donadigo) [inline src/Views/ShortcutsView.vala]: You could initialize this field in the `construct` block:
`size_group = new Gtk.SizeGroup (Gtk.SizeGroupMode.HORIZONTAL);`

File: src/Views/ShortcutsView.vala [hunk 2]
```diff
@@ -124,4 +96,17 @@ public class ShortcutOverlay.ShortcutsView : Gtk.Grid {
         add (new Gtk.Separator (Gtk.Orientation.VERTICAL));
         add (column_end);
     }
+
+    private void add_shortcut_entries (Gee.ArrayList<ShortcutEntry> entries, Gtk.Grid column) {
+        for (int a = 0; a < entries.size; a++) {
```
Comment (donadigo) [inline src/Views/ShortcutsView.vala]: Could you change `a` to `i`? It's generally a more recognizable pattern in for loops.

File: src/Views/ShortcutsView.vala (hunk 1 above)
Comment (otapliger) [reply src/Views/ShortcutsView.vala]: Done... do I need to make a new pull request or it will update the request automatically?

**Comment by cassidyjames** [comment]
Adding per-entry CSS should definitely not be needed. If we want to do something about the last item in a column, for example, that can always be done with a `:last-child` selector.

File: src/Views/ShortcutsView.vala [hunk 3]
```diff
@@ -124,4 +98,17 @@ public class ShortcutOverlay.ShortcutsView : Gtk.Grid {
         add (new Gtk.Separator (Gtk.Orientation.VERTICAL));
         add (column_end);
     }
+
+    private void add_shortcut_entries (Gee.ArrayList<ShortcutEntry> entries, Gtk.Grid column) {
+        for (int i = 0; i < entries.size; i++) {
+            var label = new ShortcutLabel (entries[i]);
+
+            if (i + 1 == entries.size) {
+                label.get_style_context ().add_class ("last-entry");
```
Comment (cassidyjames) [inline src/Views/ShortcutsView.vala]: Adding per-entry CSS should definitely not be needed. If we want to do something about the last item in a column, for example, that can always be done with a `:last-child` selector. In that case, I'd recommend switching back to a foreach for simplicity.

**Comment by cassidyjames** [comment]
@otapliger no worries! Thanks again for your contributions.

**Pull request closed without merging**